src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Dependencies.cs

```diff
@@ -0,0 +1,91 @@
+        {
+            source1RepoUri = GetRepoUrl(source1RepoName);
+            source2RepoUri = GetRepoUrl(source2RepoName);
+            targetRepoUri = GetRepoUrl(targetRepoName);
+
+            TestContext.WriteLine($"Creating test channel {testChannelName}");
+            await CreateTestChannelAsync(testChannelName);
+
+            TestContext.WriteLine("Set up build1 for intake into target repository");
+            Build build1 = await CreateBuildAsync(source1RepoUri, sourceBranch, sourceCommit, sourceBuildNumber, source1Assets);
+            await AddBuildToChannelAsync(build1.Id, testChannelName);
+
+            TestContext.WriteLine("Set up build2 for intake into target repository");
+            Build build2 = await CreateBuildAsync(source2RepoUri, sourceBranch, sourceCommit, sourceBuildNumber, source2Assets);
+            await AddBuildToChannelAsync(build2.Id, testChannelName);
+
+            ImmutableList<BuildRef> dependencies = ImmutableList<BuildRef>.Empty;
+            BuildRef buildRef1 = new BuildRef(build1.Id, true, 1);
+            dependencies = dependencies.Add(buildRef1);
+            BuildRef buildRef2 = new BuildRef(build2.Id, true, 2);
+            dependencies = dependencies.Add(buildRef2);
+
+            // Add the target build once, should populate the BuildDependencies table and calculate TimeToInclusion
+            TestContext.WriteLine("Set up targetBuild in target repository");
+            Build targetBuild1 = await CreateBuildAsync(targetRepoUri, targetBranch, targetCommit, target1BuildNumber, targetAssets, dependencies);
+            await AddBuildToChannelAsync(targetBuild1.Id, testChannelName);
+            Build newTargetBuild1 = new Build(targetBuild1.Id, targetBuild1.DateProduced, targetBuild1.Staleness, targetBuild1.Released,
+                targetBuild1.PublishUsingPipelines, targetBuild1.Commit, targetBuild1.Channels, targetBuild1.Assets, dependencies);
+
+            // Add the target build a second time, should populate the BuildDependencies table and use the previous TimeToInclusion
+            Build targetBuild2 = await CreateBuildAsync(targetRepoUri, targetBranch, targetCommit, target2BuildNumber, targetAssets, dependencies);
+            await AddBuildToChannelAsync(targetBuild2.Id, testChannelName);
+            Build newTargetBuild2 = new Build(targetBuild2.Id, targetBuild2.DateProduced, targetBuild2.Staleness, targetBuild2.Released,
+                targetBuild2.PublishUsingPipelines, targetBuild2.Commit, targetBuild2.Channels, targetBuild2.Assets, dependencies);
+
+            Build retrievedBuild1 = await MaestroApi.Builds.GetBuildAsync(targetBuild1.Id);
+            Build retrievedBuild2 = await MaestroApi.Builds.GetBuildAsync(targetBuild2.Id);
+
+            Assert.AreEqual(newTargetBuild1.Dependencies.Count, retrievedBuild1.Dependencies.Count);
+            Assert.AreEqual(newTargetBuild2.Dependencies.Count, retrievedBuild2.Dependencies.Count);
+
+            Assert.IsTrue(
```

Comment:
Some message so this doesn't just dump out "Expected: true, Actual: false" would be helpful for this very complicated assertion.

I'd like to see the assertion probably broken down a bit too, this is a fairly complex expression that will be hard to make sense of when it fails.